strawberry/types/federation/field.py

```diff
@@ -0,0 +1,35 @@
+from typing import Union
+
+from strawberry.types import StrawberryField
+
+_FIELD_FED_TYPE = Union[StrawberryField, 'StrawberryFederatedField']
```

Comment:
should this use typing.NewType?

Reply:
I'm not sure I understand why? I just didn't want to have to keep writing Union[..., ...] everywhere

Reply:
The reason each of the decorators takes both types was so that they could stack without worrying about order

Reply:
I was wondering if you could use this: https://docs.python.org/3/library/typing.html#newtype to declare the type, maybe it is not needed :)